src/transforms/globalCssToCssModule/globalCssToCssModule.ts

```diff
@@ -0,0 +1,75 @@
+import { readFileSync } from 'fs'
+import path from 'path'
+
+import { Project } from 'ts-morph'
+
+import { getCssModuleExportNameMap } from './css/getCssModuleExportNameMap'
+import { transformFileToCssModule } from './css/transformFileToCssModule'
+import { addClassNamesUtilImportIfNeeded } from './ts/classNamesUtility'
+import { getNodesWithClassName } from './ts/getNodesWithClassName'
+import { STYLES_IDENTIFIER, processNodesWithClassName } from './ts/processNodesWithClassName'
+
+interface CodemodResult {
+    css: {
+        source: string
+        path: string
+    }
+    ts: {
+        source: string
+        path: string
+    }
+}
+
+/**
+ * Convert globally scoped stylesheet tied to the React component into a CSS Module.
+ *
+ * 1) Find `.tsx` file.
+ * 2) Check if corresponding `.scss` file exists in the same folder.
+ * 3) Convert this `.scss` file into `.module.scss`.
+ * 4) Get info about CSS class names and matching export tokens.
+ * 5) Replace all matching class names with export tokens.
+ * 6) Add `classNames` import to the `.tsx` file if needed.
+ * 7) Add `.module.scss` import to the `.tsx` file.
+ *
+ */
+export function globalCssToCssModule(project: Project): Promise<CodemodResult[]> {
+    const codemodResults = project.getSourceFiles().map(async sourceFile => {
+        const filePath = sourceFile.getFilePath()
+
+        const parsedTsFilePath = path.parse(filePath)
+        const cssFilePath = path.resolve(parsedTsFilePath.dir, `${parsedTsFilePath.name}.scss`)
```

Comment:
@valerybugakov is it possible that tsx file has more than one styles file? (I mean it should be possible but do we need support in this case?)

Reply:
Also, do we handle scss @imports in style files when we getting info about CSS class names and matching export tokens?